src/declaration.c

```diff
@@ -51,6 +51,8 @@ bool checkFrameAccess(Loc loc, Scope *sc, AggregateDeclaration *ad, size_t iStar
             {
                 if (!fd->isThis() && !fd->isNested())
                     break;
+                if (FuncLiteralDeclaration *fld = fd->isFuncLiteralDeclaration())
+                    fld->tok = TOKdelegate;
```

Comment:
Or, the problem is in the function name - `checkSomething` VS `checkAndFixSomething`.